specs/flyout/flyout.md

```diff
@@ -0,0 +1,179 @@
+
+**Focus Management:** - Should follow the same process for focus management as a Dialog, primarily as the Popover's role will be 'dialog'.
+
+This includes:
+- Trap focus - optional, defaults to enabled - Focus is contained within the popover while it is visible.
+- First element focus - Focusing on the first focusable element when trap focus is enabled, but not overriding an element that is set to be focused by the author.
+- Returning focus to the target - optional, defaults to enabled, after popover is closed the focus would return to the target element.
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-popover`
+
+*Attributes*
+- `visible` - boolean, whether or not the Popover is visible, defaults to undefined, as this is controlled by the author
+- `trap-focus` - boolean, whether or not to keep focus contained inside the popover, defaults to true
+- `target` - string, the html id of the HTMLElement that the Popover is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `focus-target-on-close` - boolean, whether or not to focus back on to the _target element_ when the popover is closed, defaults to true
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic=
+- `delay` - number, the time delay before the popover is shown once triggered
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+    ** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Properties*
+- `targetElement` - the reference to the HTMLElement used as an target
+- `viewportElement` - the reference to the HTMLElement used as the viewport, required ot setup anchored region properly
+- `popoverVisible` - boolean, whether or not the Popover is visible
```

Comment:
How about `isVisible` or `open`?

Reply:
In keeping with how Tooltip has this, I think staying with popoverVisible helps clearly separate it from the `visible` attr.

Reply:
updated to flyoutVisible

Reply:
Is there a visible attr? If so, there is a visible JavaScript property. What would the difference be between a "visible" attribute and this property?